cmake/checks/check_01_cxx_features.cmake

```diff
@@ -731,28 +731,30 @@ CHECK_CXX_SOURCE_COMPILES(
 # expression, C++14 allows to call non-constexpr functions from constexpr
 # functions. Unfortunately, not all compilers obey the standard in this regard.
 #
-CHECK_CXX_SOURCE_COMPILES(
-  "
-  #define Assert(x,y) if (!(x)) throw y;
-  void bar()
-  {}
+IF(NOT CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
```

Comment:
what is the problem with MSVC here, does it pass this test but breaks at something more complicated?

Reply:
Yes, it passes the test but gives internal compiler errors if we actually use `constexpr` in the respective places.

Reply:
Ah, that's unfortunate because the way the macro works now is that MS VC will *never* work. It would be nice to actually encode the internal compiler error into a feature check itself, so that if MS comes around and fixes the issue, the cmake check will ultimately succeed. 

I think it's ok to do it this way for the moment just to make progress on the overarching issue, but it would be good if we could come back to it in the next few days and refine the macro. Do you have access to a compiler that shows these problems?

Separately: Can you explain the issue in the comment above?

Reply:
I discovered that `ddet_F_dC` (being a `constexpr` function) triggers the compiler error even if it just returns a `SymmetricTensor` object with all member functions and member variables deleted. At this point, I deactivated the macro already. Unfortunately, I couldn't reproduce that in a stand-alone test case, yet (and don't really intend to at the moment). Removing `constexpr` fixed the issue.

No, I don't have access to `MSVC` directly, but just through `appveyor`. I am happy if someone else looks into the issue separately and tries to find a reproducer we could use.

Reply:
I am not sure if a comment says much more than the code here but I can put something (after the testsuite finished).

Reply:
would be good to have something like "MSVC 2019 crashes with an internal compiler error when in a constexpr function see #9080", or something.

Reply:
@bangerth If you don't mind, I would actually prefer to add a comment in a follow-up PR to not run the tester again (if there aren't any other comments to address here).

Reply:
Okay, I will hit merge then.

Reply:
Yes, that's fine with me.